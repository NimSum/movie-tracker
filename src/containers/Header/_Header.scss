.header {
  padding: 1rem;
  background-color: $background;
  color: $main-text-color;
  text-align: right;
}

.nav {
  color: inherit;
  text-decoration: none;
  font-size: 1.1rem;
  padding: 1rem;
}