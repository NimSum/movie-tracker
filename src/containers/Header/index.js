import React, { Component } from 'react';
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'; 
import { formType, currentUser, showFavorites }  from '../../actions/index'

export class Header extends Component {
  handleClick = ({target}) => {
    target.textContent === 'Log-in' 
      ? this.props.formType('login')
      : this.props.formType('signup')
    switch(target.textContent) {
      case 'Log-out':
        this.props.logOutUser()
        break;
      case 'Favorites':
        this.props.showFavorites(true)
        break;
      default:
      this.props.showFavorites(false)
    }
  }

  render() {
    const { name, email } = this.props.currentUser;
    const userInfo = (
      <div>
        <h3>{`Yo watup ${name}`}</h3>
      </div>
    )
    return(
      <header className='header' onClick={ this.handleClick }>
        <NavLink to='/' className='nav'>
          Home 
        </NavLink>
        { name && userInfo}
        <NavLink to={`${name ? '/favorites' : '/login'}`} className='nav'>
            {`${ name ? 'Favorites' : 'Create Account'}`}
        </NavLink>
        <NavLink to={`${name ? '/' : '/login'}`} className='nav' >
            {`${ name ? 'Log-out' : 'Log-in'}`}
        </NavLink>
      </header>
    )
  }
}

export const mapStateToProps = (state) => ({
  currentUser: state.currentUser
})

export const mapDispatchToProps = dispatch => ({
  formType: (text) => dispatch(formType(text)),
  logOutUser: () => dispatch(currentUser()),
  showFavorites: (bool) => dispatch(showFavorites(bool))
})

export default connect(mapStateToProps, mapDispatchToProps)(Header);