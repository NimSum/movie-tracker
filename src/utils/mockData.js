export const mockLatestMovie = [{
  "adult": false,
  "backdrop_path": null,
  "belongs_to_collection": null,
  "budget": 0,
  "genres": [],
  "homepage": null,
  "id": 600900,
  "imdb_id": null,
  "original_language": "fr",
  "original_title": "Le Cri nu",
  "overview": "",
  "popularity": 0,
  "poster_path": null,
  "production_companies": [],
  "production_countries": [{
    "iso_3166_1": "FR",
    "name": "France"
  }],
  "release_date": "1979-01-01",
  "revenue": 0,
  "runtime": null,
  "spoken_languages": [],
  "status": "Released",
  "tagline": "",
  "title": "Le Cri nu",
  "video": false,
  "vote_average": 0,
  "vote_count": 0
}]

export const mockUser = {
  id: 1,
  name: "misskelly",
  email: "kelly@kelly.com",
  favorites: [{
    id: 2,
    movie_id: 287947,
    user_id: 1,
    title: "Shazam!",
  }]
}

export const mockMovies = [
  {
    movie_id: 299534,
    title: "Avengers: Endgame",
    poster_path: "/or06FN3Dka5tukK1e9sl16pB3iy.jpg",
    release_date: "2019-04-24",
    vote_average: 8.6,
    overview: "After the devastating events ...."
  },
  {
    movie_id: 447404,
    title: "Pokémon Detective Pikachu",
    poster_path: "/wgQ7APnFpf1TuviKHXeEe3KnsTV.jpg",
    release_date: "2019-05-03",
    vote_average: 7.1,
    overview: "In a world where people collect pocket-size monsters (Pokémon) to do battle...."
  }
]

export const mockSelectedMovie = {
  adult: false,
  backdrop_path: "/bi4jh0Kt0uuZGsGJoUUfqmbrjQg.jpg",
  belongs_to_collection: null,
  budget: 80000000,
  homepage: "http://www.shazammovie.com",
  id: 287947,
  imdb_id: "tt0448115",
  original_language: "en",
  original_title: "Shazam!",
  overview: "A boy is given the ability to become an adult superhero in times of need with a single magic word.",
  popularity: 307.286,
  poster_path: "/xnopI5Xtky18MPhK40cZAGAOVeV.jpg",
  release_date: "2019-03-23",
  revenue: 321937864,
  runtime: 132,
  status: "Released",
  tagline: "Just Say The Magic Word",
  title: "Shazam!",
  video: false,
  vote_average: 7.2,
  vote_count: 1528,
  showFavorites: false
}