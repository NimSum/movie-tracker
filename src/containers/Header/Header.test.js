//changes the formType depending on the link clicked
//
import React from 'react';
import { shallow } from 'enzyme'
import { Header, mapStateToProps, mapDispatchToProps } from '../Header';
import { mockUser } from '../../utils/mockData'


describe('Header', () => {
  let wrapper;
  let mockFormType;
  let mockLogOutUser;
  let mockShowFavorites;
  beforeEach(() => {
    wrapper = shallow(
      <Header 
        currentUser={mockUser}
        formType={mockFormType}
        logOutUser={mockLogOutUser}
        showFavorites={mockShowFavorites}
        />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
