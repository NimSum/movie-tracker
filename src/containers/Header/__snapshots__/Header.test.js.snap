// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  className="header"
  onClick={[Function]}
>
  <NavLink
    className="nav"
    to="/"
  >
    Home
  </NavLink>
  <div>
    <h3>
      Yo watup misskelly
    </h3>
  </div>
  <NavLink
    className="nav"
    to="/favorites"
  >
    Favorites
  </NavLink>
  <NavLink
    className="nav"
    to="/"
  >
    Log-out
  </NavLink>
</header>
`;
